test/helpers/sockets.rb

```diff
@@ -0,0 +1,435 @@
+# frozen_string_literal: true
+
+require 'socket'
+require 'timeout'
+
+module TestPuma
+
+  READ_TIMEOUT = 10
+
+  module SktPrepend
+    def get_body(path = nil, dly: nil, len: nil, timeout: 10)
+      fast_write get_req(path, dly: dly, len: len)
+      read_body timeout
+    end
+
+    def get_response(path = nil, dly: nil, len: nil, timeout: 10)
+      fast_write get_req(path, dly: dly, len: len)
+      read_response timeout
+    end
+
+    def read_body(timeout = 10)
+      response = read_response timeout
+      response.split("\r\n\r\n").last
+    end
+
+    def read_response(timeout = 10)
+      content_length = nil
+      response = ''.dup
+      t_st = Time.now
+      loop do
+        begin
+          chunk = read_nonblock(16_384, exception: true)
```

Comment:
let's pull the magic number out here into a constant (maybe `Puma::Test::SOME_CONST`) and name it appropriately